show/main.py

```diff
@@ -1257,26 +1283,34 @@ def table(verbose):
 @click.option('--verbose', is_flag=True, help="Enable verbose output")
 def logging(process, lines, follow, verbose):
     """Show system log"""
+    cmd, cmd0, cmd1, cmd2 = None, None, None, None
     if os.path.exists("/var/log.tmpfs"):
         log_path = "/var/log.tmpfs"
     else:
         log_path = "/var/log"
     if follow:
-        cmd = "sudo tail -F {}/syslog".format(log_path)
+        cmd = ['sudo', 'tail', '-F', '{}/syslog'.format(log_path)]
         run_command(cmd, display_cmd=verbose)
     else:
         if os.path.isfile("{}/syslog.1".format(log_path)):
-            cmd = "sudo cat {}/syslog.1 {}/syslog".format(log_path, log_path)
+            cmd0 = ['sudo', 'cat', '{}/syslog.1'.format(log_path), '{}/syslog'.format(log_path)]
         else:
-            cmd = "sudo cat {}/syslog".format(log_path)
+            cmd0 = ['sudo', 'cat', '{}/syslog'.format(log_path)]
 
         if process is not None:
-            cmd += " | grep '{}'".format(process)
+            cmd1 = ['grep', str(process)]
 
         if lines is not None:
-            cmd += " | tail -{}".format(lines)
+            cmd2 = ['tail', "-" + str(lines)]
 
-        run_command(cmd, display_cmd=verbose)
+        if cmd1 is None and cmd2 is None:
```

Comment:
>cmd1 [](http://example.com/codeflow?start=11&length=4)

You implemented a 2x2 conditions with 4 branches. This is not scalable if number of cmd increase. For example 10 cmd.

You can construct an array first, and feed the expanded array to function call. ref: [https://stackoverflow.com/a/7745986/2514803](https://stackoverflow.com/a/7745986/2514803)

Reply:
or cmdn could be defined as array elements from the beginning.